docs/route/meta.md

```diff
@@ -114,7 +118,217 @@ const meta: MetaFunction<
+  return getTask(params.tid);
+}
+
+export function meta({ data, parentsData }) {
+  let project = parentsData["routes/project/$pid"].project;
+  let task = data.task;
+  return [{ title: `${project.name}: ${task.name}` }];
+}
+```
+
+## `params`
+
+The route URL params. See [Dynamic Segments in the Routing Guide][url-params].
+
+## Gotchas with `meta` and Nested Routes
+
+Because multiple nested routes render at the same time, there is some merging that needs to happen to determine the meta tags that ultimately render. Remix gives you complete control over this merge because there is no obvious default.
+
+Remix will take the last matching route with a meta export and use that. This allows you to override things like `title`, remove things like `og:image` that the parent route added, or keep everything from the parent and add new meta for the child route.
+
+This can get quite tricky when you're new.
+
+Consider a route like `/projects/123`, there are likely three matching routes: `root.tsx`, `projects.tsx`, and `projects/$id.tsx`. All three may export meta descriptors.
+
+```tsx bad filename=app/root.tsx
+export const meta = () => {
+  return [
+    {
+      name: "viewport",
+      content: "width=device-width,initial-scale=1",
+    },
+    { title: "New Remix App" },
+  ];
+};
+```
+
+```tsx bad filename=app/routes/projects.tsx
+export const meta = () => {
+  return [{ title: "Projects" }];
+};
+```
+
+```tsx bad filename=app/routes/projects/$id.tsx
+export const meta: V2_MetaFunction = ({ data }) => {
+  return [{ title: data.project.name }];
+};
+```
+
+With this code, we will lose the `viewport` meta tag at `/projects` and `/projects/123` because only the last meta is used and the code doesn't merge with the parent.
+
+### Global Meta
+
+Nearly every app will have global meta like the `viewport` and `charSet`. We recommend using normal `<meta>` tags inside of the [root route][root-route] instead of the `meta` export so you simply don't have to deal with merging:
+
+```tsx filename=app/root.tsx lines=[12-16]
+import {
+  Meta,
+  Links,
+  Scripts,
+  Outlet,
+} from "@remix-run/react";
+
+export default function Root() {
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta
+          name="viewport"
+          content="width=device-width,initial-scale=1"
+        />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <Scripts />
+      </body>
+    </html>
+  );
+}
+```
+
+### Avoid Meta in Parent Routes
+
+You can also avoid the merge problem by simply not exporting meta that you want to override from parent routes. Instead of defining meta on the parent route, use the [index route][index-route]. This way you can avoid complex merge logic for things the title. Otherwise you will need to find the parent title descriptor and replace it with the child's title. It's much easier to simply not need to override by using index routes.
```

Comment:
```suggestion
You can also avoid the merge problem by simply not exporting meta that you want to override from parent routes. Instead of defining meta on the parent route, use the [index route][index-route]. This way you can avoid complex merge logic for things like the title. Otherwise, you will need to find the parent title descriptor and replace it with the child's title. It's much easier to simply not need to override by using index routes.
```